packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -633,9 +633,9 @@ export class ProjectService {
             await adapter.test();
         } catch (e) {
             Logger.error(`Error testing project adapter: ${e}`);
+            await adapter.destroy();
             throw e;
         } finally {
-            await adapter.destroy();
             await sshTunnel.disconnect();
```

Comment:
don't we need to move this line as well to the catch?
otherwise, the tunnel is going to close before we run the other warehouse queries.

Reply:
mmm, does that mean the tunnel might remain open forever ?

Reply:
in theory, the "parent" method should disconnect the tunnel the same way it destroys the adaptor.